integration-tests/soak/vrfv2_test.go

```diff
@@ -0,0 +1,363 @@
+	if testEnvironment.WillUseRemoteRunner() {
+		return
+	}
+
+	chainClient, err := blockchain.NewEVMClient(testNetwork, testEnvironment)
+	require.NoError(t, err)
+	contractDeployer, err := contracts.NewContractDeployer(chainClient)
+	require.NoError(t, err)
+	chainlinkNodes, err := client.ConnectChainlinkNodes(testEnvironment)
+	require.NoError(t, err)
+	chainClient.ParallelTransactions(true)
+
+	linkToken, err := contractDeployer.DeployLinkTokenContract()
+	require.NoError(t, err)
+	bhs, err := contractDeployer.DeployBlockhashStore()
+	require.NoError(t, err)
+	mf, err := contractDeployer.DeployMockETHLINKFeed(linkEthFeedResponse)
+	require.NoError(t, err)
+	coordinator, err := contractDeployer.DeployVRFCoordinatorV2(linkToken.Address(), bhs.Address(), mf.Address())
+	require.NoError(t, err)
+
+	consumer, err := contractDeployer.DeployVRFv2Consumer(coordinator.Address())
+	require.NoError(t, err)
+	err = actions.FundChainlinkNodes(chainlinkNodes, chainClient, testInputs.ChainlinkNodeFunding)
+	require.NoError(t, err)
+	err = chainClient.WaitForEvents()
+	require.NoError(t, err)
+
+	err = coordinator.SetConfig(
+		uint16(minimumConfirmations),
+		2.5e6,
+		86400,
+		33825,
+		linkEthFeedResponse,
+		vrf_coordinator_v2.VRFCoordinatorV2FeeConfig{
+			FulfillmentFlatFeeLinkPPMTier1: 1,
+			FulfillmentFlatFeeLinkPPMTier2: 1,
+			FulfillmentFlatFeeLinkPPMTier3: 1,
+			FulfillmentFlatFeeLinkPPMTier4: 1,
+			FulfillmentFlatFeeLinkPPMTier5: 1,
+			ReqsForTier2:                   big.NewInt(10),
+			ReqsForTier3:                   big.NewInt(20),
+			ReqsForTier4:                   big.NewInt(30),
+			ReqsForTier5:                   big.NewInt(40)},
+	)
+	require.NoError(t, err)
+	err = chainClient.WaitForEvents()
+	require.NoError(t, err)
+
+	err = coordinator.CreateSubscription()
+	require.NoError(t, err)
+	err = chainClient.WaitForEvents()
+	require.NoError(t, err)
+
+	err = coordinator.AddConsumer(subID, consumer.Address())
+	require.NoError(t, err, "Error adding a consumer to a subscription in VRFCoordinator contract")
+
+	actions.FundVRFCoordinatorV2Subscription(
+		t,
+		linkToken,
+		coordinator,
+		chainClient,
+		subID,
+		testInputs.SubscriptionFunding,
+	)
+
+	var (
+		encodedProvingKeys    = make([][2]*big.Int, 0)
+		nativeTokenKeyAddress string
+	)
+	for _, chainlinkNode := range chainlinkNodes {
+		vrfKey, err := chainlinkNode.MustCreateVRFKey()
+		require.NoError(t, err)
+		l.Debug().Interface("Key JSON", vrfKey).Msg("Created proving key")
+		pubKeyCompressed := vrfKey.Data.ID
+		jobUUID := uuid.New()
+		os := &client.VRFV2TxPipelineSpec{
+			Address: coordinator.Address(),
+		}
+		ost, err := os.String()
+		require.NoError(t, err)
+		nativeTokenKeyAddress, err = chainlinkNode.PrimaryEthAddress()
+		require.NoError(t, err)
+		_, err = chainlinkNode.MustCreateJob(&client.VRFV2JobSpec{
+			Name:                     fmt.Sprintf("vrf-%s", jobUUID),
```

Comment:
looks like this is being used multiple times to create jobs for diff test? can we wrap it in a function?